src/string.cr

```diff
@@ -1420,8 +1420,8 @@ class String
   # is not included in the *hash*, nothing is replaced.
   #
   # ```
-  # "hello".sub(/(he|l|o)/, {"he": "ha", "l": "la"}) # => "hallo"
-  # "hello".sub(/(he|l|o)/, {"l": "la"})             # => "hello"
+  # "hello".sub(/(he|l|o)/, {"he" => "ha", "l" => "la"}) # => "hallo"
```

Comment:
@maiha I see, I'd make a new PR extending `String#(g)sub` methods to accept `NamedTuple` along with `Hash`, WDYT @asterite ?